File: frontend/src/app/services/subscription.service.ts
import {Observable} from "rxjs";
import {subscriptionModel} from "../modules/models/subscriptionModel";
import {SubscriptionOrErrorsModel} from "../modules/models/subscriptionOrErrorsModel";
import {CustomerSubscriptionPageModel} from "../modules/models/customerSubscriptionPageModel";

export interface SubscriptionService {
  findAllSubscriptionsByCustomerId(customerId: number, page: number, amount: number): Observable<CustomerSubscriptionPageModel>;

  findAllSubscriptions(): Observable<subscriptionModel[]>;

  createNewSubscription(subscription: subscriptionModel): Observable<SubscriptionOrErrorsModel>;

 // deleteSubscriptionById(subscriptionId: number, customerId: number): Observable<void>;

  findSubscription(productId: number, customerId: number): Observable<subscriptionModel>;

  deleteSubscription(subscriptionId: number): Observable<void>;

  // findProductById(productId: string): Observable<ProductModel>;
  //
  // findProductTypes(): Observable<String[]>;
}

File: fapi/src/main/java/com/artsykov/fapi/service/SubscriptionService.java
package com.artsykov.fapi.service;

import com.artsykov.fapi.models.CustomerSubscriptionPageModel;
import com.artsykov.fapi.models.SubscriptionModel;
import com.artsykov.fapi.models.SubscriptionOrErrorsModel;

public interface SubscriptionService {
    SubscriptionOrErrorsModel saveSubscription(SubscriptionModel subscriptionModel);

    SubscriptionModel findSubscription(int productId, int customerId);

    CustomerSubscriptionPageModel findAllSubscriptionsByCustomerId(int customerId, int page, int amount);

    void deleteById(int id);
}
